addons/l10n_pk/demo/account_demo.py

```diff
@@ -0,0 +1,323 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models, Command
+from odoo.exceptions import UserError, ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountChartTemplate(models.AbstractModel):
+    _inherit = "account.chart.template"
+
+    @api.model
+    def _get_demo_data(self, company=False):
+        if company.account_fiscal_country_id.code == "PK":
+            if company.state_id:
+                return {
+                    'res.partner': self._get_l10n_pk_demo_data_partner(company),
+                    'account.move': self._get_l10n_pk_demo_data_move(company),
+                    'res.config.settings': self._get_l10n_pk_demo_data_config_settings(company),
+                }
+            else:
+                _logger.error(
+                    'Error while loading Accounting(Pakistan) demo data in the company "%s".',
+                    company.name
+                )
+        else:
```

Comment:
Why completely change the demo data?  Why not append it?

Reply:
@masm-odoo Comment still not applied.

Reply:
If we append `l10n_pk` data to the `account` module's demo data, it will attempt to post the `account` module's moves using the `l10n_pk` company. This will likely result in errors. Therefore, I don't believe appending the data is necessary.

Reply:
@masm-odoo What would be the reasons for it to result in errors?  (normally, the demo data should be generated with default taxes now)

Reply:
![image](https://github.com/user-attachments/assets/633acb93-cc50-4047-8ef9-a2e863bc4f6c)

As you can see, I am currently appending demo data instead of modifying it. The appended data includes the last three invoices, which were created in both `Your Company` and `PK Company`. To resolve this, I was updating the data instead of appending it.

Reply:
@jco-odoo It's best to override it to avoid conflicts and traceback because other model's demo data have reference to demo invoices we get from account which will be override here  and the invoices are not found further
https://github.com/odoo/odoo/blob/98c311dc3d8e6780822f5fd549c75cfee14e8923/addons/account/demo/account_demo.py#L20-L31

example- `ir.attachment` has reference to demo invoice in account which will be not found here, rather than `del` other keys it's best that we override it
https://github.com/odoo/odoo/blob/98c311dc3d8e6780822f5fd549c75cfee14e8923/addons/account/demo/account_demo.py#L427
`mail.message`
https://github.com/odoo/odoo/blob/98c311dc3d8e6780822f5fd549c75cfee14e8923/addons/account/demo/account_demo.py#L458